app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java

```diff
@@ -460,23 +461,25 @@ public void onItemClick(int position) {
      * Saves Current Image with grayscale filter
      */
     private void saveCurrentImageInGrayscale() {
+        tempImageUri = new ArrayList<>();
         try {
             File sdCard = Environment.getExternalStorageDirectory();
             File dir = new File(sdCard.getAbsolutePath() + "/PDFfilter");
             dir.mkdirs();
             Picasso picasso = Picasso.with(mActivity);
             Transformation transformation = new GrayscaleTransformation();
 
-            for (int countElements = mImagesUri.size() - 1; countElements >= 0; countElements--) {
+            for (int i = mImagesUri.size() - 1; i >= 0; i--) {
                 String fileName = String.format(getString(R.string.filter_file_name),
-                        String.valueOf(System.currentTimeMillis()), "grayscale");
+                        String.valueOf(System.currentTimeMillis()), i + "_grayscale");
                 File outFile = new File(dir, fileName);
                 String imagePath = outFile.getAbsolutePath();
-                picasso.load(new File(mImagesUri.get(countElements)))
+                picasso.load(new File(mImagesUri.get(i)))
                         .transform(transformation)
                         .into(getTarget(imagePath));
-                mImagesUri.remove(countElements);
```

Comment:
@vyankatesh24 I have already tried this

Reply:
Outside for loop check, if both are equal in size.
If `yes` replace `mImagesUri` ArrayList items
If `no` reiterate transformation process

Reply:
@Ni3verma Is this solution working?

Reply:
No. It resulted in 1k transformed photos in my storage but size of arraylists never become equal. I had to force close the app

Reply:
Do we have to use a new class variable Arraylist to store temp images?